Frameworks/Foundation/NSDictionary.mm

```diff
@@ -99,8 +99,11 @@ - (NSArray*)allObjects {
         return nil;
     }
 
-    StrongId<NSArray> keyArray = [_keyEnumerator allObjects];
-    StrongId<NSMutableArray> toRet = [NSMutableArray new];
+    StrongId<NSArray> keyArray;
+    keyArray.attach([_keyEnumerator allObjects]);
+
+    StrongId<NSMutableArray> toRet;
```

Comment:
No it shouldn't. Please see the implementation of StrongId. essentially all smart ptr types remove the * for convenience. 

---
In reply to: [73973051](https://github.com/Microsoft/WinObjC/pull/851#discussion_r73973051) [](ancestors = 73973051)